lib/Logger/Logger.cpp

```diff
@@ -695,21 +763,27 @@ void Logger::log(char const* logid, char const* function, char const* file,
       out.append("} ", 2);
     }
 
-    // meta data from log
-    LogContext::OverloadVisitor visitor(
-        [&out](std::string_view const& key, auto&& value) {
-          out.push_back('[');
-          out.append(key).append(": ", 2);
-          if constexpr (std::is_same_v<std::string_view,
-                                       std::remove_cv_t<std::remove_reference_t<
-                                           decltype(value)>>>) {
-            out.append(value);
-          } else {
-            out.append(std::to_string(value));
-          }
-          out.append("] ", 2);
-        });
-    logContext.visit(visitor);
+    {
+      READ_LOCKER(guard, _structuredParamsLock);
+      //  meta data from log
+      LogContext::OverloadVisitor visitor([&out](std::string_view const& key,
+                                                 auto&& value) {
+        if (!_structuredLogParams.contains(key.data())) {
```

Comment:
Same here, let's also pass the size of `key` when doing the lookup.

Reply:
Addressed